lib/Sema/TypeCheckDecl.cpp

```diff
@@ -7143,6 +7144,14 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
     TC.checkDeclAttributes(ICD);
   }
 
+  void visitPoundDiagnosticDecl(PoundDiagnosticDecl *PDD) {
+    if (PDD->hasBeenEmitted()) { return; }
+    PDD->markEmitted();
```

Comment:
I *suspect* this can never happen more than once per phase, so you may be able to get rid of `hasBeenEmitted`.

Reply:
I’ll try it and see what happens!

Reply:
Turns out it does happen twice. ☹️